auslib/web/common/history_all.py

```diff
@@ -0,0 +1,105 @@
+import json
+import logging
+from auslib.global_state import dbo
+from connexion import problem, request
+from sqlalchemy.sql.expression import null
+from auslib.web.common.history import HistoryHelper, get_input_dict
+from auslib.web.common.rules import get_rules
+from auslib.web.common.releases import get_releases, process_release_revisions
+from auslib.web.admin.views.permissions import UsersView, PermissionScheduledChangeHistoryView
+from auslib.web.admin.views.rules import RuleScheduledChangeHistoryView
+from auslib.web.admin.views.releases import ReleaseScheduledChangeHistoryView
+from auslib.web.admin.views.required_signoffs import ProductRequiredSignoffsHistoryAPIView, \
+    PermissionsRequiredSignoffsHistoryAPIView, ProductRequiredSignoffScheduledChangeHistoryView, \
+    PermissionsRequiredSignoffScheduledChangeHistoryView
+
+
+log = logging.getLogger(__name__)
+
+
+def _get_filters(obj, history_table):
+    input_dict = get_input_dict()
+    query = json.loads(input_dict.data)['query']
+    where = [False, False]
+    try:
+        where = [getattr(history_table, f) == query.get(f) for f in query]
+        where.append(history_table.product != null())
```

Comment:
This appears to be the line causing the tracebacks - it's not a safe assumption that product exists for all history tables.

Reply:
Yea, I've fixed that. Testing the endpoints has actually exposed some bugs I have to fix.
Thank you!